src/app/api/v1/user/update-comm-option/route.ts

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { getToken } from "next-auth/jwt";
+import { NextRequest, NextResponse } from "next/server";
+import AppConstants from "../../../../../appConstants";
+
+import {
+  getSubscriberByEmail,
+  setAllEmailsToPrimary,
+} from "../../../../../db/tables/subscribers";
+
+interface EmailUpdateCommOptionRequest {
+  communicationOption: string;
+}
+
+export async function POST(req: NextRequest) {
+  const token = await getToken({ req });
+
+  if (token) {
+    try {
+      const { communicationOption }: EmailUpdateCommOptionRequest =
+        await req.json();
+      const subscriber = await getSubscriberByEmail(token.email);
+      // 0 = Send breach alerts to the corresponding affected emails.
+      // 1 = Send all breach alerts to user's primary email address.
+      const allEmailsToPrimary = Number(communicationOption) === 1 || 0;
+      await setAllEmailsToPrimary(subscriber, allEmailsToPrimary);
+
+      return NextResponse.json({
+        success: true,
+        message: "Communications options updated",
+      });
+    } catch (e) {
+      console.error(e);
+      return NextResponse.json({ success: false }, { status: 500 });
```

Comment:
Q: Does this need a `message`?

Reply:
I don't think so, since we don't know what the cause of this error is - I don't think we want to expose internal eror messages to the outside world.